torch/distributed/__init__.py

```diff
@@ -65,3 +66,6 @@ def is_available() -> bool:
     )
 
     from .remote_device import _remote_device
+
+
+torch._C._distributed_c10d._set_debug_level()
```

Comment:
Does this mean that if I do something like

```
import torch.distributed as dist # runs __init__ that sets the debug level

os.environ["TORCH_DISTRIBUTED_DEBUG"] = "DETAIL"
init_process_group() # expect detail debugging here
```

would this use case still work? Basically a lot of internal use cases set the debug level at various points in python code (always before init process group/DDP) and expect the debug level set there to be respected. Would that still work after these changes (and can we possibly add a test if it is not covered)?

Reply:
Good catch. Nope it won't work since it parses the environment variable during process initialization. This is a fairly peculiar use of an environment variable. Let me think what I can do to keep this PR BC-compatible.